### DISCUSSION TITLE
Remeshing using output frame field

### DISCUSSION THREAD

**Discussion by Gowrimv** [BrunoLevy/geogram#137] [General]
Is it not possible to define a frame field corresponding to the desired output mesh  while remeshing? Not just anisotropy factor, but a new metric field at a point?

**Comment by BrunoLevy** [comment]
It is not impossible, but would require some work:
- 1: compute a high-dimensional isometric embedding defined from your metric, see [this reference](https://dl.acm.org/doi/10.1145/3197517.3201369)
- 2: remesh the high-dimensional surface using the Lloyd mesher in geogram. An example of curvature adaptation using 6D embedding is shown [here](https://link.springer.com/chapter/10.1007/978-3-642-33573-0_21). You will need a larger number of dimensions (I'd say 10 if you have full anisotropy tensor).

However, I confess that using a high D embedding is probably not the most efficient method for anisotropic surface meshing. You may also implement variants of Delaunay meshing, with an anisotropic definition of the cavity. There is a nice course on the topic by Frederic Alauzet, [here](https://pages.saclay.inria.fr/frederic.alauzet/cours/cea2010_V3.pdf).

**Comment by Gowrimv** [comment]
Thanks for the help. Just to clarify, geogram uses cvt in L2 norm with a
specified anisotropy for the whole mesh, right?

**Comment by BrunoLevy** [comment]
geogram's CVT can work in different modes:
- regular mode
- regular mode with a weight
- n-dimensional embedding + optional weight. Used by curvature-adapted meshing. Could be used to specify an anisotropy, but you need to implement the method in the article to compute the Nash embedding

**Comment by Gowrimv** [comment]
Ok. Thanks a lot!

**Comment by Gowrimv** [comment]
Hello,
I'm changing the  energy function to be minimised  to work in any p- norm
as given in your paper, *"l* *p* centroidal voronoi tessellation and its
applications", 2010. Can I make this change in geogram and use an n-d
embedding of current mesh to implement metric based *l* *p *CVT meshing?
Gowri
